### ISSUE TITLE
Incompleted webUI permissions?

### ISSUE THREAD

**Issue by zefanja** [linuxmuster/sophomorix4#11]
When I create a teacher and login to the webui I can't see the menu in the sidebar (see https://github.com/linuxmuster/linuxmuster-webui7/issues/19)

Not sure if this is a webui or sophomorix4 issue.

```bash
root@server:~# sophomorix-user -iv -u mm
{}
{
  "CONFIG" => {
    "ACTION" => {},
    "DEPENDS" => {},
    "MAYBE" => {},
    "ONE_OF" => {},
    "SOME_OF" => {}
  },
  "CONFIGURED" => {
    "help" => "TRUE",
    "info" => "TRUE",
    "json" => "TRUE",
    "verbose" => "TRUE"
  },
  "MODIFIER_OPTIONS" => {
    "json" => "TRUE",
    "verbose" => "TRUE"
  },
  "info" => 0,
  "json" => 0,
  "verbose" => 2
}
Option verbose is a modifier option
Option json is a modifier option
Hmmh. do not know what to do with option info
* forcing info mode
Option combinations successfully checked
Retrieving RootDSE...
   * RootDSE: DC=internal,DC=cdsc,DC=ac,DC=th
Binding with CN=administrator,CN=Users,DC=internal,DC=cdsc,DC=ac,DC=th
Testing if the Sophomorix Schema exists (Sophomorix-User)...
   * Sophomorix-Schema exists
Sophomorix::SophomorixSambaAD::AD_dns_get RootDSE: DC=internal,DC=cdsc,DC=ac,DC=th -> DNS: internal.cdsc.ac.th

################################################################################
 # Reading /usr/share/sophomorix/devel/sophomorix.ini                     # 
################################################################################
Distro-check: Ubuntu 18.04 is OK

################################################################################
 # Reading /etc/samba/smb.conf                                            # 
################################################################################

################################################################################
 # Parsing: net conf list                                                 # 
################################################################################

################################################################################
 # Asking domain passwordsettings from samba                              # 
################################################################################

################################################################################
 # Reading /usr/share/sophomorix/devel/master/sophomorix.conf.master      # 
################################################################################

################################################################################
 # Reading /etc/linuxmuster/sophomorix/sophomorix.conf                    # 
################################################################################

################################################################################
 # OK: default-school share exists                                        # 
################################################################################

################################################################################
 # OK: /etc/linuxmuster/sophomorix/default-school/school.conf             # 
################################################################################

################################################################################
 # Reading /usr/share/sophomorix/devel/master/school.conf.master          # 
################################################################################

################################################################################
 # Reading /etc/linuxmuster/sophomorix/default-school/school.conf         # 
################################################################################

################################################################################
User 1/1 in AD: mm in school default-school
CN=mm,OU=Teachers,OU=default-school,OU=SCHOOLS,DC=internal,DC=cdsc,DC=ac,DC=th
################################################################################
                   displayName: Max Mustermann                          
                            sn: Mustermann                              
                     givenName: Max                                     
      sophomorixFirstnameASCII: Max                                     
        sophomorixSurnameASCII: Mustermann                              
    sophomorixFirstnameInitial: M.                                      
      sophomorixSurnameInitial: M.                                      
           sophomorixUserToken: ---                                     
           sophomorixBirthdate: 01.01.1970                              
                sophomorixUnid: Muster!                                 
          sophomorixAdminClass: teachers                                
      sophomorixExitAdminClass: unknown                                 
          sophomorixSchoolname: default-school                          
           sophomorixAdminFile: teachers.csv                            
             sophomorixComment: created by sophomorix                   
       sophomorixFirstPassword: Muster!                                 
            sophomorixExamMode: ---                                     
--------------------------------------------------------------------------------
                sophomorixRole: teacher                                 
              sophomorixStatus: U                                       
        sophomorixCreationDate: 20180509132743.0Z                       
      sophomorixTolerationDate: 19700101000000.0Z                       
    sophomorixDeactivationDate: 19700101000000.0Z                       
            userAccountControl: 512                                     
--------------------------------------------------------------------------------
                          mail: mm@internal.cdsc.ac.th                  
           sophomorixMailQuota: ---:---:                                
 sophomorixMailQuotaCalculated: 1                                       
--------------------------------------------------------------------------------
               sophomorixQuota: default-school:---:---:new:---:         
               sophomorixQuota: linuxmuster-global:---:---:new:---:     
sophomorixCloudQuotaCalculated: ---                                     
--------------------------------------------------------------------------------
sophomorixWebuiPermissionsCalculated:
------------- sophomorixCustom: ------------------------------------------------
---------- sophomorixIntrinsic: ------------------------------------------------
--------------------------------------------------------------------------------
memberOf: CN=internet,OU=Management,OU=default-school,OU=SCHOOLS,DC=internal,DC=cdsc,DC=ac,DC=th
memberOf: CN=intranet,OU=Management,OU=default-school,OU=SCHOOLS,DC=internal,DC=cdsc,DC=ac,DC=th
memberOf: CN=printing,OU=Management,OU=default-school,OU=SCHOOLS,DC=internal,DC=cdsc,DC=ac,DC=th
memberOf: CN=teachers,OU=Teachers,OU=default-school,OU=SCHOOLS,DC=internal,DC=cdsc,DC=ac,DC=th
memberOf: CN=webfilter,OU=Management,OU=default-school,OU=SCHOOLS,DC=internal,DC=cdsc,DC=ac,DC=th
memberOf: CN=wifi,OU=Management,OU=default-school,OU=SCHOOLS,DC=internal,DC=cdsc,DC=ac,DC=th
--------------------------------------------------------------------------------
      homeDirectory: \\internal.cdsc.ac.th\default-school\teachers\homes\mm
          homeDrive: H:                                                
     accountExpires: 9223372036854775807                               
    badPasswordTime: 0                                                 
        badPwdCount: 0                                                 
         pwdLastSet: 131703464416419140                                
         lastLogoff: 0                                                 
          lastLogon: 0                                                 
         logonCount: 0                                                 
          objectSid: (binary)                                          
         objectGUID: (binary)                                          
     sAMAccountType: 805306368                                         
  userPrincipalName: mm@internal.cdsc.ac.th                            
         uSNChanged: 4040                                              
         uSNCreated: 4021                                              
           codePage: 0                                                 
        countryCode: 0                                                 
          uidNumber: 10006                                             
  unixHomeDirectory: /srv/samba/schools/default-school/teachers/homes/mm
     primaryGroupID: 513                                               
--------------------------------------------------------------------------------
LOGFILES (1 Entries):
  ADD:  2018-05-09 13:27:43  mm (Mustermann, Max) in teachers as teacher

More info for users in AD with:
   sophomorix-quota -iv -u mm
   sophomorix-mail -iv -u mm

```

**Comment by jeffbeck** [comment]
The sophomorixWebuiPermissionsCalculated are relevant for the permissions that you have as a user.
They should never be empty, but show wahat you can (true) and can't (false) do.

At the moment the calculation of sophomorixWebuiPermissionsCalculated is not integrated in sophomorix-add and update (role change-> update).

Just run:
sophomorix-ui
and the MultiValue field sophomorixWebuiPermissionsCalculated will be populated.

(The integration of sophomorix-ui) is coming soon. 
So don't close this issue.

**Comment by jeffbeck** [comment]
This is closed and tested  in 3.51.6-1:

1) sophomorix-add/update includes updating sophomorixWebuiPermissionsCalculated
2) sophomorix-ui updates sophomorixWebuiPermissionsCalculated after changes are made in school.conf/sophomorixWebuiPermissions(at the user)/webui-package upgrade
